rfcs/theme-tokens.md

```diff
@@ -0,0 +1,247 @@
+      brandForeground: brand.tint10, // different global token than in light
+    },
+  },
+};
+```
+
+### Component tokens
+
+There is nothing like component tokens in the Theme. Components use alias or global tokens directly.
+
+## Theme Overrides
+
+### For the whole application
+
+If application needs to tweak look and feel of Fluent UI components, it can change the theme for the whole application:
+
+```js
+const tweakedTheme = mergeThemes(teamsLightTheme, {
+  global: {
+    borderRadius: {
+      medium: '6px', // change medium border radius to 6px for the whole application
+    },
+  },
+});
+```
+
+This change will affect all instances of all Fluent UI components which use medium border radius in the whole application.
+
+#### For an application subtree
+
+If the application needs to tweak only its one part (screen), it will wrap that part with additional provider overriding necessary variables:
+
+```jsx
+const CustomScreen = () => {
+  // overrides only, FluentProvider will merge with a theme from context
+  const themeOverrides = {
+    global: {
+      borderRadius: {
+        medium: '6px',
+      },
+    },
+  };
+
+  return (
+    <FluentProvider theme={themeOverrides}>
+      <CustomScreenInner />
+    </FluentProvider>
+  );
+};
+```
+
+#### For an instance of a component
+
+To override tokens for a single instance of a component, you can wrap the component with a `ThemeProvider`:
+
+```jsx
+const radiusOverrides = {
+  global: {
+    borderRadius: {
+      medium: '6px'
+    }
+  }
+}
+
+<ThemeProvider theme={{radiusOverrides}}>
+  <Button>I have custom border radius</Button>
+</ThemeProvider>
+```
+
+You can also override styles using `makeStyles`:
+
+```jsx
+const useRadiusOverrides = makeStyles([[null, { borderRadius: '6px' }]]);
+
+const CustomButton = ({ className, ...rest }) => {
+  const overrides = useRadiusOverrides({});
+  const classes = ax(overrides, className);
+
+  return <Button className={classes} {...rest} />;
+};
+```
+
+#### For all instances of a single component
+
+There is no way how you could override `border-radius` for all `Button` instances in your application without altering another components. If you change `global.borderRadius.medium`, all components that use that token would be affected.
+`Button` uses this token directly, as described earlier, there is not any component token like `buttonBorderRadius` which you could override.
+
+### On-demand component tokens
+
+If engineers see that the options above do not work for their use case, they should talk to design who can add a component token to a theme.
+
+There are basically the two following rules:
+
+1. Only tokens which make sense to be tweaked will be added. We can add tokens in advance if we believe those will be eventually used, but the number of tokens should be limited.
+1. We should prefer generic tokens over specific tokens. If `componentRadiusMedium` satisfies design needs, it is better token than `buttonPrimaryTopLeftRadius`.
```

Comment:
Both these are great declarations of our points. Let's also give the `why?` part of the answer for each of them.

Reply:
The next section answers my question. Thanks.